mlir/lib/Target/LLVM/NVVM/Target.cpp

```diff
@@ -538,6 +555,30 @@ NVPTXSerializer::compileToBinaryNVPTX(const std::string &ptxCode) {
   });
 #undef DEBUG_TYPE
   RETURN_ON_NVPTXCOMPILER_ERROR(nvPTXCompilerDestroy(&compiler));
+
+  if (targetOptions.getCompilationTarget() == gpu::CompilationTarget::Fatbin) {
+    const char *cubinOpts[1] = {"-64"};
+    if (useFatbin32) {
+      cubinOpts[0] = {"-32"};
+    }
```

Comment:
Can this not be done with a ternary operator somehow? Also, is it really necessary to use `const char *` here? Using `const char *` in C++ often hints at some design flaw.

Reply:
Let me fix it. Thanks!